tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/AspNetCoreTestFixture.cs

```diff
@@ -145,24 +145,33 @@ private async Task EnsureServerStarted()
             var intervals = maxMillisecondsToWait / intervalMilliseconds;
             var serverReady = false;
 
-            // wait for server to be ready to receive requests
-            while (intervals-- > 0)
+            if (sendHealthCheck == true)
             {
-                try
+                // wait for server to be ready to receive requests
+                while (intervals-- > 0)
                 {
-                    serverReady = await SubmitRequest("/alive-check") == HttpStatusCode.OK;
-                }
-                catch
-                {
-                    // ignore
-                }
+                    try
+                    {
+                        serverReady = await SubmitRequest("/alive-check") == HttpStatusCode.OK;
+                    }
+                    catch
+                    {
+                        // ignore
+                    }
 
-                if (serverReady)
-                {
-                    break;
-                }
+                    if (serverReady)
+                    {
+                        break;
+                    }
 
+                    Thread.Sleep(intervalMilliseconds);
+                }
+            }
+            else
+            {
+                // Just wait one interval just in case
                 Thread.Sleep(intervalMilliseconds);
+                serverReady = true;
```

Comment:
I guess this has the potential to be a source of flake, if the app isn't _actually_ ready to receive requests 🤔 Doesn't seem like there's an easy answer, just something to bear in mind if we continue seeing flake in the Iast tests I think